### PR TITLE
initial reproducibility tests

### LINKED ISSUE #4: Tests for parallelism and reproducibility on simulator
-  Setting seed should result in same results parallel or not
-  Setting seed should NOT result in same results across patches or simulations
-  Also test against a model saved as RDS with same seed
-  non-seeded run after seeded run should not always be the same

See http://r-pkgs.had.co.nz/tests.html

### PR DESCRIPTION
**Pull request by calebasaraba**
closes #4

### CODE REVIEW

File: tests/testthat/test_reproducibility.R [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  expect_false(isTRUE(identical(output1a,output2a)))
+  #non-parallel check
+  output1 <- mf_sim(init = initial_cond, parameters = try_parms, times=0:1000, n_sims = 2, seed = 2, parallel = FALSE)
+  output1a <- mf_sim(init = initial_cond, parameters = try_parms, times=0:1000, n_sims = 2, parallel = FALSE)
+  output2 <- mf_sim(init = initial_cond, parameters = try_parms, times=0:1000, n_sims = 2, seed = 2, parallel = FALSE)
+  output2a <- mf_sim(init = initial_cond, parameters = try_parms, times=0:1000, n_sims = 2, parallel = FALSE)
+  expect_false(isTRUE(identical(output1a,output2a)))
+})
+
+test_that("RDS-saved model same as model created with same settings", {
```
Comment (noamross) [inline tests/testthat/test_reproducibility.R:85]: For this test I'd prefer you run the outside of this script and save the RDS in the `testthat` directory and commit that file to the repo. Then load it in this script and test against a new run.  The idea is to make sure that seeding works identically across machines and operating systems - the file you save on your local mac should produce an identical result when we run on a Linux server.

File: tests/testthat/test_reproducibility.R [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+try_parms = list(
+  beta = 0.004,                     #contact rate for direct transmission
+  gamma = 0.167,                    #recovery rate
+  mu = 0,                           #base mortality rate
+  alpha = 0,                        #disease mortality rate
+  phi = 1.96e-4,                    #infectiousness of environmental virions
+  eta = 0.14,                       #degradation rate of environmental virions
+  nu =  0.001,                      #uptake rate of environmental virion
+  sigma = 1e5,                      #virion shedding rate
```
Comment (noamross) [inline tests/testthat/test_reproducibility.R]: You should probably change sigma to zero for the purposes of these tests.

File: tests/testthat/test_reproducibility.R [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+test_that("patches are not identical", {
+  patch3_parms = list(
+    beta = 0.004,                     #contact rate for direct transmission
+    gamma = 0.167,                    #recovery rate
+    mu = 0,                           #base mortality rate
+    alpha = 0,                        #disease mortality rate
+    phi = 1.96e-4,                    #infectiousness of environmental virions
+    eta = 0.14,                       #degradation rate of environmental virions
+    nu =  0.001,                      #uptake rate of environmental virion
+    sigma = 1e5,                      #virion shedding rate
```
Comment (noamross) [inline tests/testthat/test_reproducibility.R]: Here, too.

**Review by noamross — CHANGES_REQUESTED** [review]

File: tests/testthat/test_reproducibility.R (hunk 2 above)
Comment (calebasaraba) [reply tests/testthat/test_reproducibility.R]: Done.

File: tests/testthat/test_reproducibility.R (hunk 3 above)
Comment (calebasaraba) [reply tests/testthat/test_reproducibility.R]: Done

File: tests/testthat/test_reproducibility.R (hunk 1 above)
Comment (calebasaraba) [reply tests/testthat/test_reproducibility.R:85]: I understand now ~ now fixed.

**Pull request merged**